frontend/src/components/Posts/post-plain.js

```diff
@@ -1,41 +1,101 @@
-import "./posts.css"
+import "./posts.css";
 import { useState } from "react";
 import { author_api } from "../../api/post_display_api";
+import { post_comment } from "../../api/comment_api";
 
 export default function PlainPost(data) {
-    //Get Author data -> author data is attached to data
-    
-    //Decide if shareable
-    let shareable = (data["post"]["visibility"] === "PUBLIC" || data["post"]["visibility"] === "FRIENDS") ? true : false
+  //Get Author data -> author data is attached to data
 
-    const port = window.location.port ? `:${window.location.port}` : "";
-    const authorUrl = `//${window.location.hostname}${port}/user/${(data.post.author.id ?? "").split('/').pop()}`; // allows linking to the author who wrote the post
+  //Decide if shareable
+  let shareable =
+    data["post"]["visibility"] === "PUBLIC" ||
+    data["post"]["visibility"] === "FRIENDS"
+      ? true
+      : false;
 
-    return (
-        <div>
-            <div className="message">
-                <div className="from">
-                    <h6><a href={authorUrl}>{data.post.author.displayName}</a></h6>
-                    {<img alt="author" src={data.post.author.profileImage}></img>}
-                </div>
-                <div className="postBody">
-                        {/* Will need to handle other post types here, plain for now */}
-                    <div className="content-container">
-                        <h5>{data["post"]["title"]}</h5>
-                        <div className="content line">
-                            {data["post"]["content"]}
-                        </div>
-                        </div>
-                        <div className="interaction-options">
-                            <button>like</button>
-                            {shareable && <div className="share">        {/* Only show if shareable */}
-                            <button>share</button>
-                        </div>}
-                    </div>
-                </div>
-            </div>
-            <div className="timestamp">{data["post"]["published"]}</div>
+  const port = window.location.port ? `:${window.location.port}` : "";
+  const authorUrl = `//${window.location.hostname}${port}/user/${(
+    data.post.author.id ?? ""
+  )
+    .split("/")
+    .pop()}`; // allows linking to the author who wrote the post
+
+  const [commentFieldVisibilty, setCommentFieldVisibilty] = useState(false);
+  const [comment, setComment] = useState("");
+  const [commentType, setCommentType] = useState("text/plain");
+  const submitComment = () => {
+    if (comment) {
+      post_comment(data.post.author.id, data.post.id, commentType, comment);
```

Comment:
instead of data.post.author.id we should fetch the current user to submit as the comment author